decisions/2020/06/30-frontend-peerDeps-and-releases.md

```diff
@@ -0,0 +1,94 @@
+**Date**: 2020-06-30
+
+**Attending (16)**: Austin, Birk, Christos, Edoardo, Erik, Hendrik, Ismay, Jan, Jan-Gerke, Jennifer, Joakim, Joe, Lars, Martin, Milagros, Wahed
+
+**Next Meeting**: Wednesday, August 5 @ 15:00 CET
+
+## Agenda
+
+- Proposal Ratification
+  - [PeerDeps for wrapped external dependencies](https://github.com/dhis2/notes/issues/67)
+  - [PeerDeps for Singletons](https://github.com/dhis2/notes/issues/67)
+  - [Release Strategy](https://github.com/dhis2/notes/issues/101) - slower, avoid major releases where possible, not **complete** until propagated
+  - [Jira vs. GH Issues](https://dhis2.slack.com/archives/CDX5CE15W/p1592488981276700)
+  - Managing workload and context switching (see below)
+
+## Links
+
+(DHIS2 account required)
+
+* [Meeting Recording](https://drive.google.com/file/d/1CPkBMWMy38iNu62K6FgiVYvANxmEw6mg/view)
+
+# Notes
+
+## Good News - Developer Workshop
+
+- Successfully hosted first online DHIS2 application developer workshop last week!
+- 30+ participants from 19 countries
+- Find 
+- Huge thanks to Jan-Gerke, Hendrik, Christos, Milagros, Lars, and Wahed as well as Matthieu, Alice, and Grant for making this possible!
+- Some room for improvement for 
+
+## Proposals
+
+### PeerDependencies for Wrapped External Dependencies
+
+Notably applies to `prop-types`, `final-form`, and `react-final-form`
+
+**Proposal**: Leave as-is for now, wrap and re-export dependencies from libraries that are clear extensions of an external library.
+However, plan a formal investigation into the advantages and challenges of moving to peer dependencies for a later major release (across libraries)
+
+**Status**: Approved
+
+**Next steps**: Plan investigation into implications of moving to peer dependencies in a future major version
+
+### PeerDependencies for Singletons
+
+Notably applies to `react`, `react-dom`, `styled-jsx`, `@dhis2/app-runtime`, `@dhis2/d2-i18n` (until deprecated), `@dhis2/ui` or `@dhis2/ui-core`. 
```

Comment:
Are `dhis2/ui` and children singletons? Especially confused about the core library; that should just be a collection of components. 🤔

Reply:
I'm not sure if ui has to be a singleton, but ui-core needs to be for the components in it that rely on react context. Unless we remove react context from ui-core, that'd also resolve it.

Reply:
Ah! That'd be it. Thanks for that reminder, @ismay.

Reply:
N.p. 😊👍